source/News.cpp

```diff
@@ -44,15 +44,29 @@ void News::Load(const DataNode &node, const ConditionsStore *playerConditions)
 
 		if(tag == "location")
 		{
-			if(remove)
+			if(add && !location.IsEmpty())
+				child.PrintTrace("Error: Cannot \"add\" to an existing location filter:");
+			else if(remove)
+			{
 				location = LocationFilter{};
+				if(child.HasChildren())
+					child.PrintTrace("Warning: Removing full location filter; partial removal is not supported:");
+			}
 			else
+			{
+				if(add)
+					child.PrintTrace("Warning: Cannot \"add\" to location filter, performing replace instead:");
```

Comment:
I don't think it makes sense to have this warning. Since adding can only be used on empty location filters, adding is synonymous with replacing anyway.

Reply:
I think it does make sense, if we want to ensure that the meanings of `add` and `replace` are different. But we can keep this for another PR, doesn't need to be changed today.